### DISCUSSION TITLE
ADO ExecuteReader datatype change

### DISCUSSION THREAD

**Discussion by RKvign01** [Cinchoo/ChoETL#291] [General]
Hi, 

I am trying to set datatype to Guid via WithField but it is always set as string

`                using (SqlConnection conn = new SqlConnection(StreamExtensions.getConnectionString()))
                {
                    using (Stream stream = new MemoryStream())
                    {
                        conn.Open();
                        using SqlCommand cmd = conn.CreateCommand();
                        cmd.CommandText = queryDB;
                        using (var reader = cmd.ExecuteReader())
                        {
                            using (var w = new ChoParquetWriter(stream).WithField("BuildID", fieldType: typeof(Guid)))
                            {
                                w.Write(reader);
                            }
                            byte1 = stream.ReadAllBytes();
                        }
                        conn.Close();
                        return File(byte1, "application/octet-stream", "test.parquet");
                    }
                }`

**Comment by RKvign01** [comment]
<img width="479" alt="image" src="https://github.com/Cinchoo/ChoETL/assets/112384324/bcf0d86c-3550-44a7-9b9b-3571d5079251">

**Comment by RKvign01** [comment]
I also tried this 

` using (var w = new ChoParquetWriter(stream).WithField("BuildID", typeof(Guid)))`

**Comment by RKvign01** [comment]
Find my complete program below:

[HttpGet("[Action]/{dbName}/{tbleName}")]
        public IActionResult CSVToSQL([FromRoute] string dbName = "_RH_Builds_dev", [FromRoute] string tbleName = "builds")
        {
            string queryDB = $"select * from [{dbName}].[dbo].{tbleName}";
            byte[] byte1;
            try
            {



                using (SqlConnection conn = new SqlConnection(StreamExtensions.getConnectionString()))
                {
                    using (Stream stream = new MemoryStream())
                    {
                        conn.Open();
                        using SqlCommand cmd = conn.CreateCommand();
                        cmd.CommandText = queryDB;
                        using (var reader = cmd.ExecuteReader())
                        {
                            using (var w = new ChoParquetWriter(stream).WithField("BuildID", typeof(Guid)))
                            {
                                w.Write(reader);
                            }
                            byte1 = stream.ReadAllBytes();
                        }
                        conn.Close();
                        return File(byte1, "application/octet-stream", "test.parquet");
                    }
                }
            }
            catch (Exception ex)
            {
                Console.WriteLine(ex.Message);
                return StatusCode(500, ex.Message);
            }
        }

    public static class StreamExtensions
    {
        public static byte[] ReadAllBytes(this Stream instream)
        {
            if (instream is MemoryStream)
                return ((MemoryStream)instream).ToArray();

            using (var memoryStream = new MemoryStream())
            {
                instream.CopyTo(memoryStream);
                return memoryStream.ToArray();
            }
        }
    }

**Comment by RKvign01** [comment]
Also how to handle nullable bit(bool in c#) in sql?

how to remove null values from fields?

**Comment by Cinchoo** [comment]
in regard to GUID, parquet doesn't support natively. hence ChoETL convert them to string, store them.
